sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -497,6 +498,41 @@ class Analyzer(
           ordering.map(order => resolveExpression(order, child).asInstanceOf[SortOrder])
         Sort(newOrdering, global, child)
 
+      // Replace the index with the related attribute for ORDER BY
+      // which is a 1-base position of the projection list.
+      case s @ Sort(orders, global, child) if child.resolved &&
+        orders.exists(o => IntegerIndex.unapply(o.child).nonEmpty) =>
+        val newOrders = orders map {
+          case s @ SortOrder(IntegerIndex(index), direction) =>
+            if (index > 0 && index <= child.output.size) {
+              SortOrder(child.output(index - 1), direction)
+            } else {
+              throw new UnresolvedException(s,
+                s"""Order by position: $index does not exist \n
+                    |The Select List is indexed from 1 to ${child.output.size}""".stripMargin)
+            }
+          case other => other
+        }
+        Sort(newOrders, global, child)
+
+      // Replace the index with the related attribute for Group BY
+      // which is a 1-base position of the underlying columns.
+      case a @ Aggregate(groups, aggs, child) if child.resolved &&
+        groups.exists(IntegerIndex.unapply(_).nonEmpty) =>
+        val newGroups = groups map {
+          case IntegerIndex(index) =>
+            val attributes = child.output
```

Comment:
I'm not sure if this is correct. 1 and 2 in the group by here refers to the position 1 and 2 in the select list, not the underlying query output.